DS implementations/QUEUES/Queue_Implementation.cpp

```diff
@@ -0,0 +1,108 @@
+// C++ program for array implementation of queue
+
+#include <bits/stdc++.h>
+using namespace std;
+
+// A class to represent a queue
+class Queue
+{
+public:
+    int front, rear, size;
+    unsigned capacity;
+    int *array;
+};
+
+// function to create a queue of given capacity, it initializes size of queue as 0
+Queue *createQueue(unsigned capacity)
+{
+    Queue *queue = new Queue();
+    queue->capacity = capacity;
+    queue->front = queue->size = 0;
+
+    // assigning last index of the queue to the rear.
+    queue->rear = capacity - 1;
+    // dynamically allocating int array of capacity.
+    queue->array = new int[(queue->capacity * sizeof(int))]; 
```

Comment:
You do not need to multiply with the size of datatype using `new` keyword
You are confused with `malloc` here